sky/exceptions.py

```diff
@@ -164,6 +165,12 @@ class StorageModeError(StorageSpecError):
     pass
 
 
+class StorageMountPathError(StorageSpecError):
+    # Error raised when bucket is successfully initialized, but mounting fails
+    # due to non-empty mount path
+    pass
```

Comment:
Nit: there's an issue from way before saying storage exceptions are too complex. Maybe we shouldn't add another one. How about a simple RuntimeError for now? When we have enough API usage in the wild, we can use that to decide how to design exceptions.